### PR TITLE
chore(clients/go): fix linter warnings

### LINKED ISSUE #3004: Formatting and style issues in Go client
**Describe the bug**
The Go client has some places where it's not correctly formatted, we should correct this.
Similarly, `golint` also outputs a lot of style issues which can also be easily solved. I'm not proposing that we strictly enforce `golint`, the project itself doesn't claim to be a gold standard. However, it's a useful tool and we might want to use more often.

**To Reproduce**
https://goreportcard.com/report/github.com/zeebe-io/zeebe shows the output of several compliance tools

**Expected behavior**
Go code that follows guidelines that Go developers expect.

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

  * fixes linter warnings
  * returns interface in `New...CredentialsCache` instead of unexported struct
  * technically, this PR breaks backwards compatibility but actually it's probably not a huge deal since the struct was unexported so the only way it could be stored/passed was through an implementing interface (since the methods weren't changed, these have not reason to break).
  * the remaining linter issues weren't fixed since they would break compat in the commands (likely to bother users) and they're not very serious

## Related issues

closes #3004

## Pull Request Checklist

- [ ] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [ ] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [ ] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: clients/go/pkg/zbc/oauthCredentialsCache_test.go [hunk 1]
```diff
@@ -86,7 +86,9 @@ func (s *oauthCredsCacheTestSuite) TestWriteCacheGoldenFile() {
 	err = cache.Update(capybaraAudience, capybara)
 	s.NoError(err)
 
-	cacheCopy, _ := NewOAuthYamlCredentialsCache(cachePath)
+	cacheCopyI, _ := NewOAuthYamlCredentialsCache(cachePath)
```
Comment (npepinpe) [inline clients/go/pkg/zbc/oauthCredentialsCache_test.go]: We could also have `NewOAuthYamlCredentialsCache` wrap a `newOAuthYamlCredentialsCache` which returns the unexpected struct, and us that here. IDK, I don't care that much, this is also fine for me

**Review by npepinpe — APPROVED** [review]
Good catch :+1:

**Comment by MiguelPires** [comment]
bors retry

**Pull request merged**